File: 프로그래머스/1/12926. 시저 암호/시저 암호.cpp
#include <string>
#include <vector>

using namespace std;
string solution(string s, int n) {
string answer = "";
	int i = 0;
	int j = 0;
	while (j < s.length()) {
		if (s[i] >= 'a' && s[i] <= 'z') {
			if (s[i] + n > 'z') {
				answer += s[i] + n - ('z' - 'a' + 1);
			}
			else {
				answer += s[i] + n;
			}
		}
		else if (s[i] >= 'A' && s[i] <= 'Z') {
			if (s[i] + n > 'Z') {
				answer += s[i] + n - ('Z' - 'A' + 1);
			}
			else {
				answer += s[i] + n;
			}
		}
		else if (s[i] == ' ') {
			answer += " ";
		}
        i++;
        j++;
    }
	return answer;
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.py
from typing import List

def solution(s: str, n: int) -> str:
    result = []
    for ch in s:
        if 'a' <= ch <= 'z':
            offset = (ord(ch) - ord('a') + n) % 26
            result.append(chr(ord('a') + offset))
        elif 'A' <= ch <= 'Z':
            offset = (ord(ch) - ord('A') + n) % 26
            result.append(chr(ord('A') + offset))
        else:
            result.append(ch)
    return ''.join(result)
